packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -499,6 +502,7 @@ class EditableText extends StatefulWidget {
        assert(focusNode != null),
        assert(obscuringCharacter != null && obscuringCharacter.length == 1),
        assert(obscureText != null),
+       assert(obscureTextBehavior != null),
```

Comment:
I think you can also add an assertion that both `obscureText` and `obscureTextBehavior` aren't used at the same time, to avoid confusion.